[RDY] Doc cleanup (4) - clipboard docs + various

Pyrohh opened the pull request:
I have some questions about the clipboard section in `gui_x11.txt`, which I've mostly transferred to `nvim_clipboard.txt`. I think some of it's helpful, but I'm not sure all of it's still applicable.

cc @bfredl

Commit:
Remove 'guipty' references

Presumably due to tarruda's unifdefing, it was already a no-op at the
time of nvim's first commit.

It's probably better to be clear that it doesn't exist, as opposed to
users thinking `:set guipty` is doing something when it isn't.

Commit:
doc: Remove gui_x11.txt, merge remnants into nvim_clipboard.txt

We've never had an X11 GUI.

Commit:
Remove VisualNOS highlight group remnants

Nvim hasn't ever had X11 integration, so this has was never needed.

justinmk commented on runtime/doc/nvim_clipboard.txt:
```diff
@@ -29,4 +35,77 @@ following option:
 See 'clipboard' for details and more options.
 
 ==============================================================================
+2. X11 selection mechanism				    *nvim-clipboard-x11*
```
:+1: to moving this, but the tag should remain `*x11-selection*`. I don't know why we're prefixing a bunch of tags with `nvim-`, I think we need to reconsider that practice...

Pyrohh replied:
> but the tag should remain *x11-selection*

sure

> I don't know why we're prefixing a bunch of tags with nvim-, I think we need to reconsider that practice...

I think it makes sense in certain cases, such as namespacing headers (e.g., `remote-plugin-(intro|hosts|example|manifest)`, but yeah, I agree it's not needed in this case.

Pyrohh commented:
@justinmk repushed

lucc commented:
(Sorry I commented on the commits above and not on the diff of this PR so you don't see the lines here.)

bfredl commented on runtime/doc/nvim_clipboard.txt:
```diff
@@ -29,4 +35,77 @@ following option:
+overwriting selected text.
+								*x11-cut-buffer*
+There are, by default, 8 cut-buffers: CUT_BUFFER0 to CUT_BUFFER7.
+Vim only uses CUT_BUFFER0, which is the one that xterm uses by default.
```
It don't seem that `xsel` uses the cut-buffer as a fallback. Nor I think that we want to. Instead, to keep any owned selection alive after nvim exists, we should detach the clipboard owning `xsel` job ( or alternatively relaunch `xsel` in daemon mode.)

Pyrohh replied:
> Instead, to keep any owned selection alive after nvim exists, we should detach the clipboard owning xsel job ( or alternatively relaunch xsel in daemon mode.)

Maybe, but would that introduce the possibility of multiple `xsel` daemons being active simultaneously? Anyways, I'm just updating the docs, so I'll remove this section.

bfredl replied:
: +1: we can try implement retaining clipboard when/if someone actually asks about it.

> but would that introduce the possibility of multiple xsel daemons being active simultaneously? 

No, `xsel` exits immediately when it loses ownership, no matter the process management.

bfredl commented on runtime/doc/nvim_clipboard.txt:
```diff
@@ -29,4 +35,77 @@ following option:
 ==============================================================================
+2. X11 selection mechanism		      *nvim-clipboard-x11* *x11-selection*
+
+X11 provides two basic types of global store, selections and cut-buffers,
```
I don't think this should mention cut-buffers so prominently, and give reader (unfamiliar with the inner workings of X11) the impression that they are on equal footing with selections. They are generally considered obsolete (as they don't support formats, for instance) and many modern applications simply don't use them at all.

fmoralesc commented:
@Pyrohh seconding @lucc's comments, I think it's best to move the `guipty` syntax to the vimOnlyOption group, like I did in #2902 (maybe rebase this over that?), or even leave that for now and I can add it there.

Pyrohh commented:
Thanks for taking a look everyone.

@fmoralesc @lucc

> seconding @lucc's comments, I think it's best to move the guipty syntax to the vimOnlyOption group, like I did in #2902 (maybe rebase this over that?), or even leave that for now and I can add it there.

Sure, I dropped the `syntax/vim.vim` changes, so I shouldn't need to rebase on top of your PR. The things I removed (so you can add them to your PR) were `(no|inv|)guipty` and `VisualNOS`.

Pyrohh commented:
Just rebased and fixed the merge conflict. Forgot to do it earlier, sorry.

fmoralesc commented:
OK, I added those changes to my PR ;)

Pyrohh commented:
@bfredl Thanks for taking a look, I reduced the size of the X11 clipboard section a lot. I rewrote some things so people should still have enough info to use the X11 clipboard providers, and (roughly) how they work.

If it's too sparse, please let me know.

bfredl commented:
@Pyrohh a few suggestions, otherwise LGTM :+1:

Pyrohh commented:
Okey, repushed once more with suggestions by @bfredl and @fmoralesc. I'll merge this in a day or two if no more comments.

lucc commented:
You are deleting the tags for xdefaults, athena and motif in the gui_x11.txt file. If I grep for these in the repository I get a lot of matches (not for the tags but the words). These references should maybe be cleared up in a follow up of this PR.

Pyrohh commented:
> These references should maybe be cleared up in a follow up of this PR.

Of course, I'll add it to my list.
